physics/module_bl_mynn.F90

```diff
@@ -3966,101 +3825,89 @@ SUBROUTINE  mym_condensation (kts,kte,  &
-
-          !Original C-B cloud fraction, allows cloud fractions out to q1 = -3.5
-           cldfra_bl1D(K) = MAX(0., MIN(1., 0.5+0.36*ATAN(1.55*q1(k)))) ! Eq. 7 in CB02
-           !This form only allows cloud fractions out to q1 = -1.8
-           !cldfra_bl1D(K) = MAX(0., MIN(1., 0.5+0.41*ATAN(1.55*q1(k))))
-           !This form only allows cloud fractions out to q1 = -1
-           !cldfra_bl1D(K) = MAX(0., MIN(1., 0.5+0.50*ATAN(1.55*q1(k))))
+           sgm(k) = min( sgm(k), qsat_tk*0.666 )
+           sgm(k) = max( sgm(k), qsat_tk*0.035 )
+           q1(k)  = qmq  / sgm(k)  ! Q1, the normalized saturation
+
+           !Add condition for falling/settling into low-RH layers, so at least
+           !some cloud fraction is applied for all qc and qi.
+           rh_hack = rh(k)
+           !ensure adequate RH & q1 when qi is at least 1e-9
+           if (qi(k)>1.e-9) then
+              rh_hack =min(1.0, rhcrit + 0.06*(9.0 + log10(qi(k))))
+              rh(k)   =max(rh(k), rh_hack)
+              !add rh-based q1
+              q1_rh   =-3. + 3.*(rh_hack-rhcrit)/(1.-rhcrit)
+              q1(k)   =max(q1_rh, q1(k) )
+           endif
+           !ensure adequate RH & q1 when qc is at least 1e-6
+           if (qc(k)>1.e-6) then
+              rh_hack =min(1.0, rhcrit + 0.09*(6.0 + log10(qc(k))))
+              rh(k)   =max(rh(k), rh_hack)
+              !add rh-based q1
+              q1_rh   =-3. + 3.*(rh_hack-rhcrit)/(1.-rhcrit)
+              q1(k)   =max(q1_rh, q1(k) )
+           endif
 
-        END DO
+           q1k   = q1(k)          ! backup Q1 for later modification
 
-        ! Specify hydrometeors
-        ! JAYMES- this option added 8 May 2015
-        ! The cloud water formulations are taken from CB02, Eq. 8.
-        ! "fng" represents the non-Gaussian contribution to the liquid
-        ! water flux; these formulations are from Cuijpers and Bechtold
-        ! (1995), Eq. 7.  CB95 also draws from Bechtold et al. 1995,
-        ! hereafter BCMT95
-        zagl = 0.
-        DO k = kts,kte-1
-           t    = th(k)*exner(k)
-           q1k  = q1(k)
-           zagl = zagl + dz(k)
+           ! Specify cloud fraction
+           !Original C-B cloud fraction, allows cloud fractions out to q1 = -3.5
+           !cldfra_bl1D(K) = max(0., min(1., 0.5+0.36*atan(1.55*q1(k)))) ! Eq. 7 in CB02
+           !Waynes LES fit  - over-diffuse, when limits removed from vt & vq & fng
+           !cldfra_bl1D(K) = max(0., min(1., 0.5+0.36*atan(1.2*(q1(k)+0.4))))
+           !Best compromise: Improves marine stratus without adding much cold bias.
+           cldfra_bl1D(k) = max(0., min(1., 0.5+0.36*atan(1.8*(q1(k)+0.2))))
 
-           !CLOUD WATER AND ICE
+           ! Specify hydrometeors
+           ! JAYMES- this option added 8 May 2015
+           ! The cloud water formulations are taken from CB02, Eq. 8.
            IF (q1k < 0.) THEN        !unsaturated
-#ifdef SINGLE_PREC
```

Comment:
Don't we need this `#ifdef SINGLE_PREC`?

Reply:
I'm going to guess this came from NRL (or Sam). I'm not sure of its origins. Nor did I intend to delete it. I'll get it back in just in case.